### DISCUSSION TITLE
Serving assets in development - path issue

### DISCUSSION THREAD

**Discussion by joshsharp** [martenframework/marten#159] [Q&A] [answered]
Hi Marten community! I've just got started with Marten and as someone who predominantly works with Django, I love its approach.

I feel like I must be doing something wrong because I'm pretty confused, but I am trying to serve my assets from within one folder named `static` (which is at the same level as `src`, not a child of it). However, no matter what I change, Marten seems to be looking for an `assets` folder.

In `config/routes.cr`:

```crystal
if Marten.env.development?
  puts "serving assets from #{Marten.settings.assets.root} at #{Marten.settings.assets.url}"
  path "#{Marten.settings.assets.url}<path:path>", Marten::Handlers::Defaults::Development::ServeAsset, name: "asset"
  path "#{Marten.settings.media_files.url}<path:path>", Marten::Handlers::Defaults::Development::ServeMediaFile, name: "media_file"
end
```

In `settings/development.cr`:

```crystal
Marten.configure :development do |config|
  # ...other stuff...
  
  # Assets
  config.assets.root = "static"
  config.assets.url = "/static/"
  config.assets.dirs = [
    Path["static/"]
  ]
```

Upon running `marten serve`:

```shell
serving assets from assets at /assets/
[I] [2024-02-15 03:28:49 UTC] [Server] Marten running on http://0.0.0.0:8000 (Press CTRL+C to quit)
```

Can anyone point out what silly thing I'm doing wrong? I shouldn't need to make my own `Marten::Core::Store::FileSystem` for  `storage` because it should be using my values for `config.assets.root` etc already, right? 

Thanks!

**Comment by ellmetha** [comment, marked as answer]
Hey! Thanks for trying Marten! 🙇‍♂️ 

I think I see what's happening. In your `src/project.cr` file, can you ensure that the requirement of `config/routes` is done _after_ the requirement of the settings files? 

For example:

```crystal
# Third party requirements.
require "marten"
require "sqlite3"

# Configuration requirements.
require "../config/settings/base"
require "../config/settings/**"
require "../config/initializers/**"
require "../config/routes"

# Project requirements.
# ...
```

Does it fix it?

It's definitely an issue that should be fixed though: generated projects should have the requirements in the right order. We can release a patch version with a fix promptly.

**Reply by joshsharp** [reply]
That was it, you're right! I thought it'd be something like to do with overriding the default config but didn't know where to start looking. Thanks so much for the help 🙂
